Azdevify eshopOnWeb (#844)
dotnet-architecture/eShopOnWeb: README.md

- Development Process for Azure-Hosted ASP.NET Core Apps
- Azure Hosting Recommendations for ASP.NET Core Web Apps

## Running the sample using Azd template

The store's home page should look like this:

![eShopOnWeb home page screenshot](https://user-images.githubusercontent.com/782127/88414268-92d83a00-cdaa-11ea-9b4c-db67d95be039.png)

The Azure Developer CLI (`azd`) is a developer-centric command-line interface (CLI) tool for creating Azure applications.

You need to install it before running and deploying with Azure Developer CLI.

### Windows

```powershell
powershell -ex AllSigned -c "Invoke-RestMethod 'https://aka.ms/install-azd.ps1' | Invoke-Expression"
```

### Linux/MacOS

```
curl -fsSL https://aka.ms/install-azd.sh | bash
```

> 

Reviewer: Add a link to the https://aka.ms/azure-dev/install page and mention that they can also install with package managers, like winget, choco, and brew.

After logging in with the following command, you will be able to use the azd cli to quickly provision and deploy the application.

```
azd login
```

Then, just use the `azd up` command to complete all operations of clone, provision and deployment.
```

> azd up -t dotnet-architecture/eShopOnWeb 

Reviewer: Break this into two steps, azd init -t dotnet-architecture/eShopOnWeb

and then second step

azd up

> azd up -t dotnet-architecture/eShopOnWeb -b main 

Reviewer: ```suggestion
azd up -t dotnet-architecture/eShopOnWeb
```

```
According to the prompt, enter an `env name`, and select `subscription` and `location`, these are the necessary parameters when you create resources. Wait a moment for the resource deployment to complete, click the web endpoint and you will see the home page.

**Notes:**

> Considering security, we store its related data (id, password) in the key vault when we create the database, and obtain it from the key vault when we use it. This is different from directly deploying applications locally.

Reviewer: ```suggestion
Considering security, we store its related data (id, password) in the Azure Key Vault when we create the database, and obtain it from the Key Vault when we use it. This is different from directly deploying applications locally.
```

curl -fsSL https://aka.ms/install-azd.sh | bash
```

> And you can also install with package managers, like winget, choco, and brew. For more detials, you can follow the documentation: https://aka.ms/azure-dev/install.

Reviewer: ```suggestion
And you can also install with package managers, like winget, choco, and brew. For more details, you can follow the documentation: https://aka.ms/azure-dev/install.
```

After logging in with the following command, you will be able to use the azd cli to quickly provision and deploy the application.

```
azd login
```

> Then, executes the commands `azd init` to initializes environment.

Reviewer: ```suggestion
Then, execute the `azd init` command to initialize the environment.
```

```
azd init -t dotnet-architecture/eShopOnWeb 
```

> And executes the commands `azd up` to complete all operations of provision and deployment.

Reviewer: ```suggestion
Run `azd up` to provision all the resources to Azure and deploy the code to those resources.
```

### Configuring the sample to use SQL Server

> 1. In the `src/Web/Program.cs` file, switch to the `Locally` configuration of SQL Server.

Reviewer: @zedy-wj - Can you please update the app so that no code changes are required to run this locally?

Author: @jongio - We added a parameter in [appSettings](https://github.com/dotnet-architecture/eShopOnWeb/pull/844/files#diff-7ef659fc9cf6968e718894d300490b14ea7a52091e7d4bcffae3a5029ac721d4R63) to distinguish different environments of `locally` and `Azd template`.
